isso/tests/test_comments.py

```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
         self.app.db.comments.activate(1)
         self.assertEqual(self.client.get('/?uri=test').status_code, 200)
 
+    def testModerateComment(self):
+
+        id_ = 1
+        signed = self.app.sign(id_)
+
+        # Create new comment, should have mode=2 (pending moderation)
+        rv = self.client.post(
+            '/new?uri=moderated', data=json.dumps({"text": "..."}))
+        self.assertEqual(rv.status_code, 202)
+        self.assertEqual(self.client.get('/id/1').status_code, 200)
+        self.assertEqual(self.app.db.comments.get(id_)["mode"], 2)
+        self.assertEqual(self.app.db.comments.get(id_)["text"], "...")
+
+        # GET should return some html form
+        action = "activate"
+        rv_activat_get = self.client.get('/id/%d/%s/%s' % (id_, action, signed))
+        self.assertEqual(rv_activat_get.status_code, 200)
+        self.assertIn(b"Activate: Are you sure?", rv_activat_get.data)
+        self.assertIn(b"http://invalid.localmoderated#isso-1", rv_activat_get.data)
+
+        # GET should return some html form
```

Comment:
Any reason for duplicating this test?

Reply:
nope this is a rebase mistake i overlooked